src/datachain/lib/udf_signature.py

```diff
@@ -58,15 +58,23 @@ def parse(
         if not isinstance(udf_func, UDFBase) and not callable(udf_func):
             raise UdfSignatureError(chain, f"UDF '{udf_func}' is not callable")
 
-        func_params_map_sign, func_outs_sign, is_iterator = (
-            UdfSignature._func_signature(chain, udf_func)
+        func_params_map_sign, func_outs_sign, is_iterator = cls._func_signature(
+            chain, udf_func
         )
+
+        udf_params: dict[str, Union[DataType, Any]] = {}
         if params:
-            udf_params = [params] if isinstance(params, str) else params
-        elif not func_params_map_sign:
-            udf_params = []
-        else:
-            udf_params = list(func_params_map_sign.keys())
+            udf_params = (
+                {params: Any} if isinstance(params, str) else dict.fromkeys(params, Any)
+            )
```

Comment:
Use `Any` type here for params with unknown type, since we don't have an `Unknown` type in Python 👀

Reply:
can it be None? what is parameter type is type annotation is missing ... e.g. how does mypy treat it?

Reply:
This is the case when `params` is set in `map` chain method. Here we are not checking UDF signature param types, e.g.:

```python
dc.map(lambda file: file.parse_info(), params="file")
```

or

```python
def process_signals(signal1, signal2) -> int:
    return signal1 + signal2

dc.map(process_signals, params=["signal1", "signal2"])
```

---

Answering your question, if parameter type is missing, `inspect` will return it as `inspect.Parameter.empty` (see [below](https://github.com/iterative/datachain/pull/973/files#diff-5d3be55eec360ab345279685e953f961a442346fd180ddfd1df214af2a96eccaR71-R76)). In this case we are returning it's type as `Any` (since where is no `Unknown` type in Python).